unblob/handlers/archive/zip.py

```diff
@@ -80,18 +80,16 @@ def _calculate_zipfile_end(self, file: io.BufferedIOBase, start_offset: int) ->
         self.cparser_le.end_of_central_directory_t(file)
         return file.tell()
 
-    def is_valid(self, file: io.BufferedIOBase) -> bool:
+    def is_valid(self, file: io.BufferedIOBase) -> Tuple[bool, bool]:
```

Comment:
One function should do one thing at a time. This should be split into two: `is_valid` and `has_encrypted_files`

Reply:
I would not split this function. This is already going through the whole file to check if it is valid and it is encrypted, it just did not return with the encryption info.

Instead, I renamed it. Now calling it looks clear enough:


```python
    is_valid, has_encrypted_files = self.check_file(file)
```